app/src/main/java/com/example/paritosh/stormy/WeatherContract.java

```diff
@@ -0,0 +1,34 @@
+package com.example.paritosh.stormy;
+
+
+import android.content.Context;
+import android.support.annotation.StringRes;
+
+import com.example.paritosh.stormy.model.CurrentWeatherDataBindingModel;
+
+public interface WeatherContract {
+
+    interface WeatherView {
+
+        void render(CurrentWeatherDataBindingModel model);
+
+        void showMessage(@StringRes int messageResId);
+
+        void showApiFailError();
```

Comment:
void showApiFailError(); and void showConnectivityError(); can be squashed into one method void showErrorMessage(@StringRes int titleResId, @StringRes int messageResId);

Then it can be implemented in the activity like,
showErrorMessage(@StringRes int titleResId, @StringRes int messageResId) {
SabkaAlertDialogFragment.newInstance(
                titleResId,
                messageResId
        ).show(getSupportFragmentManager(), "error dialog");
}